solana/programs/token-bridge-relayer/src/processor/inbound.rs

```diff
@@ -300,25 +287,94 @@ fn token_bridge_complete_wrapped(
             token_program: ctx.accounts.token_program.to_account_info(),
             wormhole_program: ctx.accounts.wormhole_program.to_account_info(),
         },
-        signer_seeds,
+        &[redeemer_seeds],
     ))
 }
 
-fn is_native(ctx: &Context<CompleteTransfer>) -> TokenBridgeRelayerResult<bool> {
-    let check_native = create_native_check(ctx.accounts.mint.mint_authority);
+/// If the transfer includes a gas dropoff, the gas will be transferred to the recipient.
+fn redeem_gas(ctx: &Context<CompleteTransfer>, gas_dropoff_amount: u32) -> Result<()> {
+    // Denormalize the gas_dropoff_amount.
+    // Since it is transferred as Klamports, we need to convert it to lamports:
+    let gas_dropoff_amount = u64::from(gas_dropoff_amount) * 1_000;
```

Comment:
Solano and I hacked in the * 1000 while trying to get transfers to work. It should at least get a constant that explains wtf it does (namely, gas dropoff amount is specified in klamports and there has to be transformed so that a value of 1 becomes 1000 lamports)